src/Distribution/Server/Features/Search/ExtractDescriptionTerms.hs

```diff
@@ -78,7 +78,7 @@ termsMarkup = Markup {
   markupBold          = id,
   markupMonospaced    = \s -> if length s > 1 then [] else s,
   markupUnorderedList = concat,
-  markupOrderedList   = concat,
+  markupOrderedList   = concat . map snd,
```

Comment:
This won't work for `haddock-library-1.10.0`.  You will also have to bump the lower bound of `haddock-library` to `>= 1.11.0` if you want to sort this out without CPP directives like `#if MIN_VERSION_haddock_library(1,11,0)`.